www-reputation-credentials-protocol.md

```diff
@@ -0,0 +1,265 @@
+is uniform in its outcome (success/failure) across all the intermediary counterparties, however such counterparties
+cannot influence the outcome, once the chain of contract is established. As setting up this chain of contracts
+consume scarce resources (i.e the channel liquidity timevalue), there is a counterparty risk without guarantee
+there is an adequate compensation. Indeed, the routing fees are only paid in case of success.
+
+# Solution introduction
+
+This counterparty risk can be solved straightforwardly by the introduction of a premimum fee carried on by
+the HTLC sender. This jamming solution known as ["upfront fees"](https://github.com/t-bast/lightning-docs/blob/master/spam-prevention.md)
+in the Lightning community, aims to a fair distribution of the HTLC forward risk among all the chain of
+contracts counterparties. While there is a concern this risk distribution doesn't capture well the
+shenanigans of a HTLC forward (e.g offline routing hop, spontaneously congestioned channel), there is a
+major drawback as it introduces a permanent overhead fee for the network participants.
+
+In contract, another type of reputation-based solution can be introduced by formalizing a new assumption
+"Reputation incentives HTLC senders to behave-well, if there is an adequate proportion between the
+acquisition cost and the amount of resources credited in function of said reputation". This assumption is
+in fact analoguous to the one underpining routing scoring algorithms incentiving the routing hops reliability.
+A reputation system attached to the HTLC forward, not only can protect against malicious jamming entity but
+also can offer the traffic shapping tooling for node operators to improve the quality of their inbound HTLC
+across time.
+
+# Proposed solution
+
+The proposed solution is to introduce a new reputation credentials system, where credentials
+should be attached to each HTLC forward request. The credentials are issued locally by each
+routing hop, there is no global credential system maintained for the whole Lightning Network.
+The credentials represent channel liquidity lockup rights, covering both amount and CLTV duration.
+The exact ratio between a quantity of credentials and the channel liquidity lockup rights is
+a routing hop policy decision and can be adjusted in real-time in function of channel congestion
+rate and other factors. This routing policy can be announced towards the LN network through
+the public gossips mechanism.
+
+The credentials are disseminated from the routing hops towards the HTLC senders through onion
+communication channels. The credentials acquisition method and cost are defined by the routing
+hops. For this version of the protocol, we propose upfront fees as a credential acquisition
+method. Payment HTLC are sent to the routing hop and credentials are returned back to the sender
+with a onion reply path. The credentials are counter-signed by the issuer to authenticate their
```

Comment:
>The credentials are counter-signed by the issuer to authenticate their origin.

First I thought that a secret (without a signature) should be sufficient. But then I realized, because of onions, you get a signature "for free", right?

Reply:
The rational of the signature, it allows credentials authentication by the routing hop, otherwise one is free to forge credential to cover HTLC routing requests. The signature means "This credential has been endorsed by myself during a previous round in its blinded version". I can see how using the phrasing "authenticate their origin" is confusing, should say more to endorse their "endorsement".

Reply:
But how this is different from issuing a secret only the issuer and the receiver knows?

Reply:
I understand, if the issuer can map the secret to a given receiver, you loss privacy of the credentials. Blinded signature is a simple scheme providing unlinkability. We might have other cryptosystems to consider.